### PR TITLE
Update stacklayout.py; Fixes issue 4504

### PR DESCRIPTION
**Pull request by JackAnderson5**
Fixes [issue 4504](https://github.com/kivy/kivy/issues/4504). 

`lc =[]` is irrelevant; simply moved closer to where it's used.

### CODE REVIEW

**Comment by tshirtman** [comment]
i'd really like a better solution, but unless someone comes up with a better idea, it'll have to get merged…

**Comment by matham** [comment]
The only suggestion I have is to not actually change the size of the child as this pr does using `sizes`, but instead only use that `1e-12` trick when checking if ti fits. I'm worried that e.g. on a 640 width with two children instead of each child's size being 320, it'll now be `320 - 1e-12` which may have other effects, but will also make it harder to display the number cleanly.

**Comment by user5061** [comment]
Sorry for the mess. Didn't know how to post it here. There [you go](https://github.com/kivy/kivy/pull/4579).

<sub>Offtopic: Original issue with StackLayout and original fix are raised by my old account</sub>

**Comment by user5061** [comment]
TL;DR: `1e-12` should be fine. (Much smaller values could be a problem though.)

Initially i used `1e-9` just to be on the safe side, since i didnt know up to how many digits a float is "precise". For example `101*.2=20.200000000000003`, error occurs on 15th decimal so subtracting `1e-15` would be a bad idea. However, with `2201*.8=1760.8000000000002`, the error is on 13th decimal.

**Comment by dessant** [comment]
Closing in favor of https://github.com/kivy/kivy/pull/4579.
